### PR TITLE
Support case sensitive Snowflake identifier [sc-10531]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

For case-sensitive snowflake table names, we need to put them in double quotes when formulating extra_info query. Otherwise, it's throwing invalid query exception.

### 🤓 What?

- use double quotes when formulating extra_info query

### 🧪 Tested?

tested against snowflake DB with a case-sensitive table name.

### CODE REVIEW

File: metaphor/snowflake/utils.py [hunk 1]
```diff
@@ -37,6 +37,11 @@ class QueryWithParam:
     params: Optional[Tuple] = None
 
 
+def to_quoted_identifier(parts: List[Optional[str]]) -> str:
```
Comment (mars-lan) [inline metaphor/snowflake/utils.py:40]: Test?

File: metaphor/common/entity_id.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def normalize_full_dataset_name(name: str) -> str:
     """
     Normalizes a fully qualified dataset name
     """
-    return name.lower().replace("`", "")
+    return name.lower().replace("`", "").replace('"', "")
```
Comment (mars-lan) [inline metaphor/common/entity_id.py:84]: Consider adding a test.

**Review by mars-lan — COMMENTED** [review]
Thanks for the quick fix.

File: metaphor/common/entity_id.py (hunk 2 above)
Comment (alyiwang) [reply metaphor/common/entity_id.py:84]: added

File: metaphor/snowflake/utils.py (hunk 1 above)
Comment (alyiwang) [reply metaphor/snowflake/utils.py:40]: added

**Review by mars-lan — APPROVED** [review]

**Review by mars-lan — APPROVED** [review]

**Pull request merged**